enzyme/Enzyme/TypeAnalysis/TypeTree.h

```diff
@@ -248,7 +247,8 @@ class TypeTree : public std::enable_shared_from_this<TypeTree> {
             if (CT == BaseType::Anything || CT == pair.second) {
               // previous equivalent values or values overwritten by
               // an anything are removed
-              toremove.insert(pair.first);
+              changed = true;
```

Comment:
Is this legal here (std map iirc), for some std containers I remember erasure invalidates all iterators.

Reply:
> References and iterators to the erased elements are invalidated. Other references and iterators are not affected.
https://en.cppreference.com/w/cpp/container/map/erase